src/TMF8801/TMF8801_Api.h

```diff
@@ -75,6 +75,7 @@ class TMF8801_Api
   void application_readObjectDetectionResult(ObjectDetectionData & data);
   void application_loadCalibData(CalibData const & calib_data);
   void application_loadAlgoState(AlgoState const & algo_state);
+  void application_read_serial_number(uint32_t const & serial_number);
```

Comment:
Why `const &`? Does this even work? I suggest the following signature ...
```suggestion
  uint32_t application_read_serial_number();
```